service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java

```diff
@@ -328,13 +328,11 @@ public Object run() throws HiveSQLException {
             LOG.error("Error running hive query: ", e);
           } finally {
             LogUtils.unregisterLoggingContext();
-            Hive hiveDb = Hive.getThreadLocal();
-            if (hiveDb != null && hiveDb != parentHive) {
-              // If new hive object is created  by the child thread, then we need to close it as it might
-              // have created a hms connection. Call Hive.closeCurrent() that closes the HMS connection, causes
-              // HMS connection leaks otherwise.
```

Comment:
When I create sessionHive, I set the allowClose flag to false. So, even if I try to invoke close on sessionHive via closeCurrent, it will be no-op and doesn't close it. So, this check is not needed.

Reply:
i am not sure if we can safely assume that BackgroundWork is always called with session hive